src/ModelContextProtocol.AspNetCore/StreamableHttpHandler.cs

```diff
@@ -108,27 +133,36 @@ public async Task HandleDeleteRequestAsync(HttpContext context)
 
     private async ValueTask<HttpMcpSession<StreamableHttpServerTransport>?> GetSessionAsync(HttpContext context, string sessionId)
     {
-        if (Sessions.TryGetValue(sessionId, out var existingSession))
+        HttpMcpSession<StreamableHttpServerTransport>? session;
+
+        if (HttpServerTransportOptions.Stateless)
         {
-            if (!existingSession.HasSameUserId(context.User))
-            {
-                await WriteJsonRpcErrorAsync(context,
-                    "Forbidden: The currently authenticated user does not match the user who initiated the session.",
-                    StatusCodes.Status403Forbidden);
-                return null;
-            }
+            var sessionJson = Protector.Unprotect(sessionId);
+            var statelessSessionId = JsonSerializer.Deserialize(sessionJson, StatelessSessionIdJsonContext.Default.StatelessSessionId);
+            var transport = new StreamableHttpServerTransport();
+            session = await CreateSessionAsync(context, transport, sessionId, statelessSessionId);
+        }
+        else if (!Sessions.TryGetValue(sessionId, out session))
+        {
+            // -32001 isn't part of the MCP standard, but this is what the typescript-sdk currently does.
```

Comment:
Should we add it to McpErrorCode anyway? Should we work with the MCP spec folks to try to make it official, with a set of code beyond the core JSON RPC ones that are standardized for MCP?